scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1182,8 +1182,28 @@ def cmake_assemble_args(extra_args, handler, extra_conf_files, extra_overlay_con
         return args_expanded
 
     def cmake(self, filter_stages=[]):
+        args = []
+        for va in self.testsuite.extra_args.copy():
+            cond_args = va.split(":")
+            if cond_args[0] == "arch" and len(cond_args) == 3:
+                if self.instance.platform.arch == cond_args[1]:
+                    args.append(cond_args[2])
+            elif cond_args[0] == "platform" and len(cond_args) == 3:
+                if self.instance.platform.name == cond_args[1]:
+                    args.append(cond_args[2])
+            elif cond_args[0] == "simulation" and len(cond_args) == 3:
+                if self.instance.platform.simulation == cond_args[1]:
+                    args.append(cond_args[2])
+            else:
+                if cond_args[0] in ["arch", "platform", "simulation"]:
+                    logger.warning("extra_args lead with %s," +
+                        "but has no enough param exp 3 but %d",
+                        cond_args[0], len(cond_args))
```

Comment:
The warning itself is a bit confusing, keep it simple?

```suggestion
                    logger.warning("Unexpected extra_args: {va}")
```